test/unit/CookiesStorageTests/cookieSave_test.js

```diff
@@ -0,0 +1,52 @@
+goog.module('measurementLibrary.storage.CookiesStorage.testing.cookieSave');
```

Comment:
I know the PR description notes that it's not possible to directly validate cookie settings (expiry, domain) were set correctly, and that's true (at least without significant additional tooling like Selenium tests).

That said, I have a suggestion for a medium sized tweak that would allow at least some weaker tests to be written.  However, it's a decent change and I'm fine if you'd like to merge this PR without it and consider it for a followup PR.

There are a few tricks that can be used to at least record the full "write" string (so, including expiry, domain, and other settings) for indirect verification.  While there's still no way you can verify that settings are interpreted correctly by the browser, you can at least verify that your code generates set strings that *should* be generated correctly on *most* browsers.  Since the cookie API is so stable, this is still a pretty good outcome.

On to the tricks.  My favorite approach is to have a dead-simple "setCookie(cookieString)" method that can be replaced with a spy for unit tests.  So e.g.:

```
save(...) {
  // Build cookie string as in PR.
  setCookie(cookie);
}

/** @param {string} cookieStr */
setCookie(cookieStr) {
  document.cookie = cookieStr;
}
```

Then in unit tests, setCookie can be replaced with a spy and you can assert that it was called with an expected string.

Reply:
I like this idea and I think it's important to include in this PR. I looked into it and I think it would be nice use this:
spyOn(storage, "setCookie").and.callThrough();
storage.save(...);
const correctCookieString = ...;
expect(storage.setCookie).toHaveBeenCalledWith(correctCookieString);

Is this the kind of spy you think would work?

Reply:
Yeah, that's pretty much exactly what I had in mind.  The fact that Jasmine's spys make it easy to call through so you can verify at least the part of the actual behavior that you can with the cookie API is a nice perk, though!